tests/Unit/Domain/Services/TalkRating/TalkRatingContextTest.php

```diff
@@ -23,53 +25,49 @@
 class TalkRatingContextTest extends \PHPUnit\Framework\TestCase
 {
     /**
-     * @test
+     * @dataProvider strategyProvider
      */
-    public function getTalkStrategyReturnsATalkRatingStrategy()
+    public function testGetTalkRatingStrategyReturnsCorrectInstance($input, $expectedClassName)
     {
-        $strategy = TalkRatingContext::getTalkStrategy('yesno', $this->authMock());
-        $this->assertInstanceOf(TalkRatingStrategy::class, $strategy);
+        $strategy = TalkRatingContext::getTalkStrategy($input, $this->authMock());
+        $this->assertInstanceOf($expectedClassName, $strategy);
     }
 
-    /**
-     * @test
-     */
-    public function getTalkStrategyWithYesNoReturnsYesNoRating()
+    public function strategyProvider(): array
     {
-        $strategy = TalkRatingContext::getTalkStrategy('yesno', $this->authMock());
-        $this->assertInstanceOf(YesNoRating::class, $strategy);
+        return [
+            'Input is empty string' =>
+                ['', YesNoRating::class],
+            'Input is yesno' =>
+                ['yesno', YesNoRating::class],
+            'Input is onetoten' =>
+                ['onetoten', OneToTenRating::class],
+            'Casing of yesno doesnt matter' =>
+                ['YeSNo', YesNoRating::class],
+            'Casing of onetoten doesnt matter' =>
+                ['OnEToTen', OneToTenRating::class],
+            'Giberish defaults to YesNo' =>
```

Comment:
I im in favour of removing that behaviour from the factory, what do you think?

Reply:
Do you mean forcing the correct capitalisation?

Reply:
I would suggest to have no default behaviour - either a value is passed in which we can map, or we throw an `InvalidArgumentException` or something.

What do you think?

Reply:
I prefer having the default behaviour keep everything the way it currently is. That way when someone upgrades, they won't have to change their current config files. And everything keeps working.